### PR TITLE
Modernization

### PR DESCRIPTION
**Pull request by MaddTheSane**
These changes are to help make the code more future-proof, and integrate with Swift better.

### CODE REVIEW

**Comment by gnachman** [comment]
Wow.

Thank you for taking this on. It will probably take me some time to get through this all.

File: sources/AdvancedWorkingDirectoryWindowController.h [hunk 1]
```diff
@@ -14,6 +14,6 @@
 @property(nonatomic, copy) Profile *profile;
 
 // An array of keys that may be changed in self.profile. All take string values.
-- (NSArray *)allKeys;
+@property (readonly, strong) NSArray<NSString*> *allKeys;
```
Comment (gnachman) [inline sources/AdvancedWorkingDirectoryWindowController.h]: Why strong? IIRC that's ARC only and not needed for a readonly property.

Comment (MaddTheSane) [inline sources/AdvancedWorkingDirectoryWindowController.h]: Ah, you are right.
I was trying to make it more ARC and Swift friendly.

Should I revert it?

**Comment by gnachman** [comment]
I've started a branch **MaddTheSane-modernization** where I'm making some changes for style. There's a lot of testing to do, but it looks good so far.

**Comment by gnachman** [comment]
Merged. Thanks!

**Pull request closed without merging**